File: OLSource1/v1/vs140/codesnippet/CSharp/how-to--extend-a-query-by-using-code_1.cs
        partial void TopNSalesOrders_PreprocessQuery
            (short? TopN, ref IQueryable<Customer> query)
        {
            query = (from myCustomer in query
                     where myCustomer.Orders.Count() > 0
                     orderby myCustomer.Orders.Count() descending
                     select myCustomer).Take(System.Convert.ToInt16(TopN));
        }

File: OLSource1/v1/vs140/codesnippet/VisualBasic/how-to--extend-a-query-by-using-code_1.vb
        Private Sub TopNSalesOrders_PreprocessQuery _
            (ByVal TopN As System.Nullable(Of Short), _
             ByRef query As System.Linq.IQueryable(Of LS_Queries_VB.Customer))
            query = From myCustomer In query
                   Where myCustomer.Orders.Count > 0
                   Select myCustomer
                   Order By myCustomer.Orders.Count Descending
                   Take (TopN)
        End Sub